services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/mapper/MappingConfigurationTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.mapper;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static software.amazon.awssdk.enhanced.dynamodb.internal.EnhancedClientUtils.getMappingConfiguration;
+import static software.amazon.awssdk.enhanced.dynamodb.mapper.AttributeMapping.SHALLOW;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class MappingConfigurationTest {
+
+    MappingConfiguration configuration;
+    @BeforeEach
+    public void setup() {
+        configuration = getMappingConfiguration(true, SHALLOW);
+    }
+
+    @Test
+    public void mappingConfigurationTest_verify_initialization_isNotNull() {
+        assertNotNull(configuration);
+    }
+
+    @Test
+    public void mappingConfigurationTest_verify_ToString() {
+        assertEquals("MappingConfiguration(attributeMapping=SHALLOW, ignoreNulls=true)", configuration.toString());
+    }
+
+    @Test
+    public void mappingConfigurationTest_verify_equals() {
```

Comment:
nit: we can use `equalsverifier` to verify equals and hashcode. Example: https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3/src/test/java/software/amazon/awssdk/services/s3/crt/S3CrtHttpConfigurationTest.java#L24